snap/quota/resources_builder.go

```diff
@@ -93,6 +120,20 @@ func (rb *ResourcesBuilder) Build() Resources {
 			Limit: rb.ThreadLimit,
 		}
 	}
+	if rb.JournalNamespaceSet || rb.JournalSizeLimitSet || rb.JournalRateSet {
+		quotaResources.Journal = &ResourceJournal{}
```

Comment:
the case where we do this but the subfields are unset probably needs a dedicated test as it's different from all other cases, also there is not limit involved in that case

Reply:
True, I'll add a new test for this